datastore/src/main/scala/org/allenai/datastore/Datastore.scala

```diff
@@ -127,6 +127,9 @@ class Datastore(val name: String, val s3: AmazonS3Client) extends Logging {
     }
   }
 
+  /** Common base class for all datastore exceptions, so they can be caught together */
+  class Exception(message: String, cause: Throwable) extends scala.Exception(message, cause)
```

Comment:
I can see where you're coming from, and I made the change, but it means that to reference this class, you have to say `Datastore.DatastoreException`, which is redundant. When faced with a choice between making something (in this case, the `Datastore` class) nice on the inside, and making it nice on the outside, I always choose the outside. There is a lot more outside than inside.